scripts/format_arpa_lm.py

```diff
@@ -44,6 +46,9 @@
 ngram_order = int(f.readline())
 f.close()
 
+# set the memory restriction for "sort"
+sort_mem_opt = ("--buffer-size={0} ".format(args.max_memory))
```

Comment:
If 'sort' doesn't accept things like 0.5G, then you may have to have your
code turn things like 10G and 5M into numbers in bytes, and deal with those
throughout the rest of the program.


On Thu, Aug 11, 2016 at 8:37 PM, Ke Li <notifications@github.com> wrote:

> In scripts/format_arpa_lm.py
> <https://github.com/danpovey/pocolm/pull/46#discussion_r74537956>:
>
> > @@ -44,6 +46,9 @@
> >  ngram_order = int(f.readline())
> >  f.close()
> >
> > +# set the memory restriction for "sort"
> > +sort_mem_opt = ("--buffer-size={0} ".format(args.max_memory))
>
> I see your point. I set the empty case as default (no memory limitation)
> which I think is reasonable(?). I tested when --max-memory option is empty
> and it worked.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/danpovey/pocolm/pull/46/files/c42a9232ccf73be3005330a3fb811af9c2f523e9#r74537956>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVu_9TLvZrFKKJVv3_umEFoxU12tCZks5qe-p7gaJpZM4JibxT>
> .
>